sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -1058,37 +1059,66 @@ private static void validateMethodAnnotations(
     }
 
     // Verify that there is no getter with a mixed @JsonIgnore annotation.
-    validateGettersNoMixedAnnotations(
-        methodNameToAllMethodMap, descriptors, AnnotationPredicate.JSON_IGNORE);
+    validateGettersHaveConsistentAnnotation(
+        methodNameToAllMethodMap, descriptors, AnnotationPredicates.JSON_IGNORE);
 
     // Verify that there is no getter with a mixed @Default annotation.
-    validateGettersNoMixedAnnotations(
-        methodNameToAllMethodMap, descriptors, AnnotationPredicate.DEFAULT_VALUE);
+    validateGettersHaveConsistentAnnotation(
+        methodNameToAllMethodMap, descriptors, AnnotationPredicates.DEFAULT_VALUE);
 
     // Verify that no setter has @JsonIgnore.
-    validateSettersNoAnnotations(
-        methodNameToAllMethodMap, descriptors, AnnotationPredicate.JSON_IGNORE);
+    validateSettersDoNotHaveAnnotation(
+        methodNameToAllMethodMap, descriptors, AnnotationPredicates.JSON_IGNORE);
 
     // Verify that no setter has @Default.
-    validateSettersNoAnnotations(
-        methodNameToAllMethodMap, descriptors, AnnotationPredicate.DEFAULT_VALUE);
+    validateSettersDoNotHaveAnnotation(
+        methodNameToAllMethodMap, descriptors, AnnotationPredicates.DEFAULT_VALUE);
   }
 
   /**
    * Validates that getters don't have mixed annotation.
    */
-  private static void validateGettersNoMixedAnnotations(
+  private static void validateGettersHaveConsistentAnnotation(
       SortedSetMultimap<Method, Method> methodNameToAllMethodMap,
       List<PropertyDescriptor> descriptors,
-      AnnotationPredicate annotationPredicate) {
-    List<InconsistentlyAnnotatedGetters> incompletelyAnnotatedGetters = new ArrayList<>();
-    for (PropertyDescriptor descriptor : descriptors) {
+      final AnnotationPredicates annotationPredicates) {
+    List<InconsistentlyAnnotatedGetters> inconsistentlyAnnotatedGetters = new ArrayList<>();
+    for (final PropertyDescriptor descriptor : descriptors) {
       if (descriptor.getReadMethod() == null
           || IGNORED_METHODS.contains(descriptor.getReadMethod())) {
         continue;
       }
+
       SortedSet<Method> getters = methodNameToAllMethodMap.get(descriptor.getReadMethod());
-      SortedSet<Method> gettersWithTheAnnotation = Sets.filter(getters, annotationPredicate);
+      SortedSet<Method> gettersWithTheAnnotation =
+          Sets.filter(getters, annotationPredicates.forMethod);
+      Set<Annotation> propertyAnnotations = Sets.newLinkedHashSet(FluentIterable
+          .from(gettersWithTheAnnotation)
+          .transform(new Function<Method, Annotation>() {
+            @Nullable
+            @Override
+            public Annotation apply(@Nonnull Method method) {
+              FluentIterable<Annotation> methodAnnotations = FluentIterable
```

Comment:
Instead of doing the filtering here, why not chain this into the FluentIterable above

.from(...)
.transformAndConcat(Method -> Iterable<Annotation>)
.filter(annotationPredicates.forAnnotation)

would also allow you to drop the 'final' declaration on annotationPredicates